lib/tpm2_options.c

```diff
@@ -279,23 +279,30 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv, char **envp,
      * grep -rn case\ \'[a-zA-Z]\' | awk '{print $3}' | sed s/\'//g | sed s/\://g | sort | uniq | less
      */
     struct option long_options [] = {
-        { "tcti",          required_argument, NULL, 'T' },
         { "help",          no_argument,       NULL, 'h' },
         { "verbose",       no_argument,       NULL, 'v' },
         { "quiet",         no_argument,       NULL, 'Q' },
         { "version",       no_argument,       NULL, 'V' },
         { "enable-errata", no_argument,       NULL, 'Z' },
+        /*
+         * tcti must be last, because when a tool requests that
+         * a sapi is not provided, we subtract 1 from the size
+         * of the array length so --tcti is not valid.
+         */
+        { "tcti",          required_argument, NULL, 'T' },
```

Comment:
Even when are nicely documented, I always have the feeling that making these kind of assumptions is fragile. I would rather prefer to just keep all the common options but instead just check for `flags & TPM2_OPTIONS_NO_SAPI` in the `'T'` case. I've proposed PR #906 that does this if you want to evaluate that option.

Yes, that means that the tools that don't support `-T, --tcti` will still says that an argument is required if a user just passes `-T` with no argument. But it will print an error if the option is attempted to use correctly.

The option wouldn't even be documented in the tools' man pages so why an user will try to use it in the first place?

Reply:
1. The docs are lacking
2. Users do all sorts of stuff
3. When TCTI is disabled, I want the missing -T and --tcti messages thrown by getopt.